HyperPlay-Gaming/hyperplay-desktop-client
# [TECH] Add more logs and error handling for MW Mod support

- Adds a timeout while waiting for the files to be copied to the upper directory, this is necessary since some people are getting stuck forever on this step.
- Adds some permission checks to the source and destination files.
- Adds some try/catches to throw errors while extracting, copying, moving files, etc.
- Adds some tests to the `copyRecursiveAsync` method.
- Log more verbosely.
- Needed to update jest config and dependencies to handle the creating and deleting of files and folders and to deal with the timeout for the method above.

### HOW TO TEST

- Windows only
- Install Market Wars as usual to see if it will fail or suceed.
- These changes won't change the logic of the patching, only add error handlers.

---

Use the following Checklist if you have changed something on the Backend or Frontend:
